### PR TITLE
Replace default webhooks with // +kubebuilder:default annotations 

### LINKED ISSUE #4204: Use `// +kubebuilder:default` annotations where possible
/kind cleanup
/good-first-issue
/help

**What needs cleanup:**

There are several places in the defaulting webhooks where CAPZ defaults a static value that could be replaced with the [`// +kubebuilder:default`](https://book.kubebuilder.io/reference/markers/crd-validation.html) annotation on the API type.

e.g. this code defaults a static value
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/e2bd59253c859b151ac6f3222ad6c749e2a1ba75/api/v1beta1/azuremanagedcontrolplane_webhook.go#L75-L78

It should be removed and replaced by the following change to this block:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/e2bd59253c859b151ac6f3222ad6c749e2a1ba75/api/v1beta1/azuremanagedcontrolplane_types.go#L118-L123

```diff
diff --git a/api/v1beta1/azuremanagedcontrolplane_types.go b/api/v1beta1/azuremanagedcontrolplane_types.go
index 5d83db207..35adbf3db 100644
--- a/api/v1beta1/azuremanagedcontrolplane_types.go
+++ b/api/v1beta1/azuremanagedcontrolplane_types.go
@@ -126,6 +126,7 @@ type AzureManagedControlPlaneSpec struct {
 	// Allowed values are "azure", "kubenet".
 	// Immutable.
 	// +kubebuilder:validation:Enum=azure;kubenet
+	// +kubebuilder:default:=azure
 	// +optional
 	NetworkPlugin *string `json:"networkPlugin,omitempty"`
```

Tests that verify the removed code should also be removed.

After modifying the annotations, run `make generate-manifests` to encode the new defaults in the CRDs.

**Describe the solution you'd like**
Walk through the webhooks starting at the following places and replace hand-rolled code to default fields with static fields with the kubebuilder annotation:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/e2bd59253c859b151ac6f3222ad6c749e2a1ba75/api/v1beta1/azuremachine_webhook.go#L221
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/e2bd59253c859b151ac6f3222ad6c749e2a1ba75/api/v1beta1/azuremachinetemplate_webhook.go#L145
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/e2bd59253c859b151ac6f3222ad6c749e2a1ba75/api/v1beta1/azuremanagedcontrolplane_webhook.go#L70
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/e2bd59253c859b151ac6f3222ad6c749e2a1ba75/api/v1beta1/azuremanagedmachinepool_webhook.go#L68
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/e2bd59253c859b151ac6f3222ad6c749e2a1ba75/exp/api/v1beta1/azuremachinepool_webhook.go#L59

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by SubhasmitaSw**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4204

### CODE REVIEW

**Comment by nojnhuh** [comment]
Thank you @SubhasmitaSw! Could you please take a look at the build failures?

File: api/v1beta1/azuremachinetemplate_types.go [hunk 1]
```diff
@@ -29,6 +29,7 @@ type AzureMachineTemplateSpec struct {
 // +kubebuilder:object:root=true
 // +kubebuilder:resource:path=azuremachinetemplates,scope=Namespaced,categories=cluster-api
 // +kubebuilder:storageversion
+// +kubebuilder:default:Default=true
```
Comment (nawazkh) [inline api/v1beta1/azuremachinetemplate_types.go:32]: I am a little puzzled, what is this annotation meant to do?

File: api/v1beta1/azuremanagedmachinepool_types.go [hunk 2]
```diff
@@ -427,6 +427,7 @@ type AzureManagedMachinePoolSpec struct {
 
 	// Name - name of the agent pool. If not specified, CAPZ uses the name of the CR as the agent pool name.
 	// Immutable.
+	// +kubebuilder:default=default
```
Comment (nawazkh) [inline api/v1beta1/azuremanagedmachinepool_types.go:430]: I am not so sure this is right. We should probably not set a default in here.

File: exp/api/v1beta1/azuremachinepool_types.go [hunk 3]
```diff
@@ -342,6 +342,7 @@ type (
 	// +kubebuilder:subresource:status
 	// +kubebuilder:resource:path=azuremachinepools,scope=Namespaced,categories=cluster-api,shortName=amp
 	// +kubebuilder:storageversion
+	// +kubebuilder:default:=true
```
Comment (nawazkh) [inline exp/api/v1beta1/azuremachinepool_types.go:345]: I am curious to know the effects of this annotation.

File: exp/api/v1beta1/azuremachinepool_webhook.go [hunk 4]
```diff
@@ -55,15 +55,6 @@ type azureMachinePoolWebhook struct {
...
 
-// Default implements webhook.Defaulter so a webhook will be registered for the type.
-func (ampw *azureMachinePoolWebhook) Default(ctx context.Context, obj runtime.Object) error {
-	amp, ok := obj.(*AzureMachinePool)
-	if !ok {
-		return apierrors.NewBadRequest("expected an AzureMachinePool")
-	}
-	return amp.SetDefaults(ampw.Client)
-}
-
```
Comment (nawazkh) [inline exp/api/v1beta1/azuremachinepool_webhook.go:66]: Was this removal necessary?

**Review by nawazkh — COMMENTED** [review]
Good changes, but I had some questions.

**Comment by nojnhuh** [comment]
@SubhasmitaSw Are able to follow up on the build failures?

**Comment by SubhasmitaSw** [comment]
@nojnhuh Apologies for stretching this long, should get all reviews resolved this week.

**Comment by nojnhuh** [comment]
Closing this for now. Feel free to reopen this PR when you get back to it.
/close
